<!-- KITcar launch script for tof_sensor_emulation -->
<launch>

	<arg name="start_activated" default="True" />
	<arg name="name"/>
	<arg name="topic"/>

  <node ns="gazebo_simulation" pkg="gazebo_simulation" type="tof_sensor_emulation" output="screen" name="tof_sensor_emulation_node_$(arg name)">

   <rosparam file="$(find gazebo_simulation)/param/tof_sensor_emulation/default.yaml" command="load" ns="default" />
   <rosparam file="$(find gazebo_simulation)/param/tof_sensor_emulation/free_ride_mode.yaml" command="load" ns="free_ride_mode" />
   <rosparam file="$(find gazebo_simulation)/param/tof_sensor_emulation/obstacle_mode.yaml" command="load" ns="obstacle_mode" />
   <rosparam file="$(find gazebo_simulation)/param/tof_sensor_emulation/parking_mode.yaml" command="load" ns="parking_mode" />

   <param name="frame_id" value="ir_$(arg name)" />
   <param name="sub_topic" value="/camera/depth_raw/$(arg name)_points" />
   <param name="pub_topic" value="/controller_interface/controller_interface/$(arg topic)" />
   <param name="start_activated" value="$(arg start_activated)" />
  </node>
</launch>
